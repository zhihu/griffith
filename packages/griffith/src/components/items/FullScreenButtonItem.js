import React from 'react'
import {css} from 'aphrodite/no-important'
import {ua} from 'griffith-utils'
import styles from '../Controller.styles'
import Icon from '../Icon'
import * as icons from '../icons/controller'
import Tooltip from '../Tooltip'
import Hover from '../Hover'

const {isMobile} = ua

const FullScreenButtonItem = ({isFullScreen, onClick}) => (
  <Hover className={css(styles.menuContainer)}>
    {isFullScreenHovered => (
      <React.Fragment>
        <button className={css(styles.button)} onClick={onClick}>
          <Icon icon={isFullScreen ? icons.smallscreen : icons.fullscreen} />
        </button>
        <div
          className={css(
            styles.fullScreenTooltip,
            styles.menu,
            isFullScreenHovered && styles.menuShown,
            isFullScreen && styles.fullScreenTooltipWide
          )}
        >
          {!isMobile && (
            <Tooltip
              content={
                isFullScreen
                  ? 'action-exit-fullscreen'
                  : 'action-enter-fullscreen'
              }
            />
          )}
        </div>
      </React.Fragment>
    )}
  </Hover>
)

export default FullScreenButtonItem
