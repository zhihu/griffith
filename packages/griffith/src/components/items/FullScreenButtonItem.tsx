import React from 'react'
import * as icons from '../icons/controller'
import ControllerButton from './ControllerButton'
import ControllerTooltip from './ControllerTooltip'

const FullScreenButtonItem: React.FC<{
  isFullScreen: boolean
  onClick: React.HTMLAttributes<HTMLButtonElement>['onClick']
}> = ({isFullScreen, onClick}) => (
  <ControllerTooltip
    localeKey={
      isFullScreen ? 'action-exit-fullscreen' : 'action-enter-fullscreen'
    }
    hotkey="f"
  >
    <ControllerButton
      icon={isFullScreen ? icons.smallscreen : icons.fullscreen}
      onClick={onClick}
    />
  </ControllerTooltip>
)

export default FullScreenButtonItem
