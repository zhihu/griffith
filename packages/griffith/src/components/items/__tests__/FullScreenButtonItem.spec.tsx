import React from 'react'
// @ts-expect-error ts-migrate(7016) FIXME: Could not find a declaration file for module 'enzy... Remove this comment to see the full error message
import {mount} from 'enzyme'
import FullScreenButtonItem from '../FullScreenButtonItem'

describe('FullScreenButtonItem', () => {
  it('get FullScreenButtonItem component', () => {
    const onClick = jest.fn()
    const wrapper = mount(
      <FullScreenButtonItem isFullScreen={true} onClick={onClick} />
    )
    expect(wrapper).toMatchSnapshot()
    wrapper.find('button').simulate('click')
    expect(onClick).toBeCalled()
  })
})
