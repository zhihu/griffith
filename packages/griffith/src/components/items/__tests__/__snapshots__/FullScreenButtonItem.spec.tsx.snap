// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullScreenButtonItem get FullScreenButtonItem component 1`] = `
<div>
  <div class="menuContainer_e296pg">
    <button aria-label="Exit Fullscreen" class="button_1j0k413" type="button">
      <span class="root_s9vkyf"
        ><svg viewBox="0 0 24 24" class="svg_11u3rwe">
          <path
            d="M14.2884629,14.2931716 C14.1097429,14.4722267 14,14.7186945 14,14.990778 L14,20 L16,20 L16,16 L20,16 L20,14 L14.990778,14 C14.7223751,14 14.4759332,14.1106894 14.2959709,14.2898855 L14.2884629,14.2931716 Z M16,8 L16,4 L14,4 L14,9.00922203 C14,9.28130552 14.1097429,9.52777327 14.2884629,9.70682845 C14.4759332,9.88931059 14.7223751,10 14.990778,10 L20,10 L20,8 L16,8 Z M8,16 L8,20 L10,20 L10,14.990778 C10,14.7186945 9.89025712,14.4722267 9.71153715,14.2931716 C9.52406685,14.1106894 9.27762494,14 9.00922203,14 L4,14 L4,16 L8,16 L8,16 Z M9.71153715,9.70682845 C9.89025712,9.52777327 10,9.28130552 10,9.00922203 L10,4 L8,4 L8,8 L4,8 L4,10 L9.00922203,10 C9.27762494,10 9.52406685,9.88931059 9.70402912,9.71011452 L9.71153715,9.70682845 Z"
          ></path></svg
      ></span>
    </button>
    <div class="menu_1a7s4a3-o_O-tooltipContent_me14v4">
      Exit Fullscreen (f)
    </div>
  </div>
</div>

`;
